### DISCUSSION TITLE
Can't generate LetsEncrypt certificate for Vaultwarden behind nginx reverse proxy

### DISCUSSION THREAD

**Discussion by LorenzBung** [dani-garcia/vaultwarden#4264] [Q&A] [answered]
Hi everyone,
I run vaultwarden in a docker container behind nginx as reverse proxy (which is run on my machine directly) using a slightly modified version of the [suggested configuration](https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples):
```nginx
# vaultwarden
server {
        server_name vault.example.de;
        listen 80;
        listen [::]:80;
        listen 443 ssl http2;
        listen [::]:443 ssl http2;
        ssl_certificate /etc/letsencrypt/live/example.de/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/example.de/privkey.pem;

        location /.well-known/acme-challenge/ {
                root /var/www/letsencrypt;
        }
        location / {
                proxy_pass https://localhost:8555;
                proxy_set_header Host $host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
        }
}
```
I have a LetsEncrypt certificate and didn't have any problems, but today I got a mail telling me my certificate had to be renewed.
However, `sudo certbot` gives me the following error:
```
Certbot failed to authenticate some domains (authenticator: nginx). The Certificate Authority reported these problems:
  Domain: vault.example.de
  Type:   unauthorized
  Detail: ...: Invalid response from http://vault.example.de/.well-known/acme-challenge/...: 422
```
Does anybody have an idea what's the reason for this behaviour?
It might be me being just dumb :)
Thanks!

**Comment by Masgalor** [comment, marked as answer]
Your logs say you are using the `nginx authenticator` but your config says you are using `manual mode`.
I'm not sure if this is really the problem here, but if you use the nginx plugin you don't need to specify a location for the challenge files.
The plugin will dynamically reload the config as needed, your location rule `location /.well-known/acme-challenge/` might collide with that.

Anyway, as this problem is not related to vaultwarden, so you might get better help at the [letsencrypt support forum](https://community.letsencrypt.org/).

**Reply by LorenzBung** [reply]
You're correct, I fixed it. For everyone interested, I set a wildcard record in my Cloudflare DNS settings that resolved `_acme-challenge.example.de` to my local server instead of solving the challenge. I fixed it using [acme-dns-certbot](https://www.digitalocean.com/community/tutorials/how-to-acquire-a-let-s-encrypt-certificate-using-dns-validation-with-acme-dns-certbot-on-ubuntu-18-04) and set a fixed CNAME on `_acme-challenge.example.de`.
